docs/DeveloperGuide.md

```diff
@@ -383,6 +383,45 @@ The sequence diagram is as follows:
 
 The exported file can be subsequently imported back into any other instance of ContaX, similar to the existing `.json` system of import/export.
 
+### Enhanced Find Logic
+
+The previous implementation find target is limited to name field only. Hence, extending the find function will increase the user experience by enable them easily search `Person` with `phone`
+`email` and `address` field.
+
+This is done by creating `searchType` model within the `model util` package and extracting the original `NameConstrainsKeywordsPreicate` to form `ConstrainsKeywordsPreicate` then create different inherit of three different types of predicate.
+
+The sequence diagram is as follows:
+![Enhanced Find Logic](images/FindCommandSequenceDiagram.png)
+
+### Enhanced Command Features
+
+#### Chain Command
+
+This extension allows user to chain multiple commands together.
+
+:information_source: **Note:** A command failure may lead to following commands being invalid (e.g. if create person failed, you cannot edit the newly created person), causing an exception to be thrown.
+
+The sequence diagram is as follows:
+![Chain Command](images/ChainCommandSequenceDiagram.png)
+
+#### Range Command
+
+This extension allows user to perform range of commands base on `index`. Hence, `from/INDEX to/INDEX` is essential in parsing the value of input. The validation of index is similar to original command (i.e. the index in edit command).
```

Comment:
Why did you use hence here? There is no previous part to hence from, and the second sentence does not make sense.

Reply:
```suggestion
This extension allows user to perform range of commands based on `index`. Hence, `from/INDEX to/INDEX` is essential in parsing the value of input. The validation of index is similar to original command (i.e. the index in edit command).
```

Reply:
thx updated, I tried to change sencond sentence